projects/web/src/components/scan.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from "react";
+import { Link } from "react-router-dom";
+
+import { AuthState } from "../auth";
+import { Organisation } from "../business-registry";
+import { LabelField } from "./epcis/common";
+
+const { DigitalLink } = require("digital-link.js");
+
+export interface SubmitProps {
+  authState: AuthState;
+  organisation: Organisation;
+  scanData: string;
+}
+
+const DigitalLinkPrefix = "https://scan.mirza.d61.io/";
+
+export function Scan(props: SubmitProps) {
+  let epc = "";
+  let errorStr: string = null;
+
+  const dlUrl = DigitalLinkPrefix + props.scanData;
+
+  try {
```

Comment:
Not sure why you need a `try/catch` block here?

Reply:
Because the DigitalLink constructor throws exceptions on bad input.